Clarify usage of Verifier and Relying Party

aj-stein-nist opened the pull request:
Closes #120.

SteveLasker reviewed (changes requested):
Thanks, @aj-stein-nist for trying to clarify Client, Verifier and Relying Party. 
Referencing [RATS Terminology](https://www.rfc-editor.org/rfc/rfc9334.html#name-roles), the Relying Party is a Consumer while an Attestor is the Producer of information. All interactions with the Transparency Service require a Client, which almost always requires access rights to write, and may require access rights to read. 
I'd suggest keeping Client, and making Verifier and Relying Party synonymous, or replace Verifier with Relying Party.

aj-stein-nist commented on draft-ietf-scitt-architecture.md:
```diff
@@ -188,10 +188,10 @@ Auditor:
 : an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service.
 
-Client:
+Relying Party:
```
This PR is still in draft, but I intend to argue here, mailing list, and heaven forbid editor meetings if invited that if client is vague, we should use a precise term and build the necessary semantics around it (while avoiding the implementation-specific details not part of the architecture).

aj-stein-nist commented on draft-ietf-scitt-architecture.md:
```diff
@@ -553,9 +553,9 @@ Multiple Issuers can make the same Statement about a single Artifact, affirming
 To register a Signed Statement, the Transparency Service performs the following steps:
 
-1. **Client authentication:** A Client authenticates with the Transparency Service, to Register Signed Statements.
+1. **Relying Party authentication:** A Relying Party authenticates with the Transparency Service, to Register Signed Statements.
```
Part of the original report from a community reviewer and implementer is that "client" as-is is vague and not precise. Casual use of client, in addition to more precision about what this Relying Party does, seems very important to clarifying this person's understanding of the architecture. Same goes for me as well. 😄

SteveLasker replied:
I believe the added definition of Client solves the gap. As the Issuer isn't the identity that interacts with the Transparency Service, we need a way to describe that entity. A Verifier, Relying Party and Auditor all interact with the Transparency Service through a client. The client is what is authorized to write/read to/from the TS.

aj-stein-nist commented on draft-ietf-scitt-architecture.md:
```diff
@@ -388,7 +388,7 @@ Registration Policies refer to additional checks over and above the Mandatory Re
 Transparency Services MUST maintain Registration Policies which govern whether or not a given Signed Statement is eligible for registration.
 
-Registration Policies MUST be made transparent and available to all clients of the Transparency Service by registering them as Signed Statements on the Append-only Log.
+Registration Policies MUST be made transparent and available to all Relying Parties of the Transparency Service by registering them as Signed Statements on the Append-only Log.
```
I am saying we replace the term client with Relying Party like other standards.

Casual usage of client has not helped thus -far, and we should be more specific with a precise term. Additionally, the SCITT architecture is not always a traditional client-server architecture (more complex real-world use cases will function as a multi-layered distributed system), so implying that with use of the word client can further confuse unless we plan to address that.

SteveLasker replied:
Yes, by giving a 👍, I was agreeing with you in this context Relying Party was a more narrow clarification 😄

aj-stein-nist commented on draft-ietf-scitt-architecture.md:
```diff
@@ -578,7 +578,7 @@ The multiple receipts may be attached to the unprotected header of the Signed St
 ## Transparent Statements {#Receipt}
 
-The client (which is not necessarily the Issuer) that registers a Signed Statement and receives a Receipt can produce a Transparent Statement by adding the Receipt to the Unprotected Header of the Signed Statement.
+The Relying Party (which is not necessarily the Issuer) that registers a Signed Statement and receives a Receipt can produce a Transparent Statement by adding the Receipt to the Unprotected Header of the Signed Statement.
```
Relying Party will be synonymous with client/Client as we use it today, not Verifier or Issue or other entity.

aj-stein-nist commented on draft-ietf-scitt-architecture.md:
```diff
@@ -833,7 +833,7 @@ All contents exchanged between actors is protected using secure authenticated ch
 
 The Transparency Service is trusted with the confidentiality of the Signed Statements presented for Registration.
 Some Transparency Services may publish every Signed Statement in their logs, to facilitate their dissemination and auditing.
```
I'll defer on that one.

ad-l commented:
Generally the PR is straightforward (replacing Client with RP), but I am not sure there is consensus on using "relying party", as the usage of the term is fairly removed from the definition in other RFCs.

From RFC7521:
>   The entity that creates and signs or integrity-protects the assertion
>   is typically known as the "Issuer", and the entity that consumes the
>   assertion and relies on its information is typically known as the
>   "Relying Party".  In the context of this document, the authorization
>   server acts as a relying party.

the client in SCITT does not consume the statement/assertion, so in my opinion the closest analogy to Relying Party in SCITT would be the verifier, not the client.

From RFC9334:
>In Remote ATtestation procedureS (RATS), one peer (the "Attester") produces believable information about itself ("Evidence") to enable a remote peer (the "Relying Party") to decide whether or not to consider that Attester a trustworthy peer. Remote attestation procedures are facilitated by an additional vital party (the "Verifier").

If we translate this to SCITT, Attester = Issuer, Evidence = Statement, Relying Party = Verifier, Verifier = Transparency Service.
So I would understand and probably agree to use "Relying Party" as a substitute for "Verifier" but using it as a replacement for "Client" is just more confusing to me

aj-stein-nist commented:
> From RFC7521:
> 
> > The entity that creates and signs or integrity-protects the assertion
> > is typically known as the "Issuer", and the entity that consumes the
> > assertion and relies on its information is typically known as the
> > "Relying Party".  **In the context of this document, the authorization
> > server acts as a relying party.**

I added emphasis but maybe it is unclear what is "the software or tool" proxying an entity and what requirements are similar are dissimilar for Issuers versus Auditors versus Verifiers.

> From RFC9334:
> 
> > In Remote ATtestation procedureS (RATS), one peer (the "Attester") produces believable information about itself ("Evidence") to enable **a remote peer (the "Relying Party") to decide whether or not to consider that Attester a trustworthy peer. Remote attestation procedures are facilitated by an additional vital party (the "Verifier").**

I added emphasis again, but this appears to be talking about software by term "remote peer" which I would use describe software/tool, not a person/organizational entity with a role.

So on that note,

> If we translate this to SCITT, Attester = Issuer, Evidence = Statement, Relying Party = Verifier, Verifier = Transparency Service. So I would understand and probably agree to use "Relying Party" as a substitute for "Verifier" but using it as a replacement for "Client" is just more confusing to me

I am confused how we think Verifier is both "client" and "Transparency Service" at the same time. Can you elaborate further?

OR13 commented on draft-ietf-scitt-architecture.md:
```diff
@@ -188,10 +188,10 @@ Auditor:
 
-: an application making protected Transparency Service resource requests on behalf of the resource owner and with its authorization.
-The term "Client" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other devices).
+: a client application for protected Transparency Service resource requests on behalf of the resource owner and with its authorization.
```
```suggestion
: a relying party depends on or is a Verifier of Signed or Transparent Statements.
```

SteveLasker replied:
Personally, I'd like to keep CLient as defined and add Relying Party. I'd decline this suggestion, here.

OR13 commented on draft-ietf-scitt-architecture.md:
```diff
@@ -388,7 +388,7 @@ Registration Policies refer to additional checks over and above the Mandatory Re
 Transparency Services MUST maintain Registration Policies which govern whether or not a given Signed Statement is eligible for registration.
 
-Registration Policies MUST be made transparent and available to all clients of the Transparency Service by registering them as Signed Statements on the Append-only Log.
+Registration Policies MUST be made transparent and available to all Relying Parties of the Transparency Service by registering them as Signed Statements on the Append-only Log.
```
```suggestion
Registration Policies MUST be made transparent and available to all Relying Parties of the Transparency Service by registering them as Signed Statements on the Append-only Log, and distributing associated Receipts. Distribution of Receipts is out of scope for this document.
```

OR13 commented on draft-ietf-scitt-architecture.md:
```diff
@@ -553,9 +553,9 @@ Multiple Issuers can make the same Statement about a single Artifact, affirming
 To register a Signed Statement, the Transparency Service performs the following steps:
 
-1. **Client authentication:** A Client authenticates with the Transparency Service, to Register Signed Statements.
+1. **Relying Party authentication:** A Relying Party authenticates with the Transparency Service, to Register Signed Statements.
```
```suggestion
1. **Client authentication:** A client application authenticates with the Transparency Service, to Register Signed Statements on behalf of one or more issuers.
```

OR13 commented on draft-ietf-scitt-architecture.md:
```diff
@@ -553,9 +553,9 @@ Multiple Issuers can make the same Statement about a single Artifact, affirming
+1. **Relying Party authentication:** A Relying Party authenticates with the Transparency Service, to Register Signed Statements.
 Authentication and authorization is implementation-specific, and out of scope of the SCITT Architecture.
-1. **Issuer Verification:** The Transparency Service MUST perform resolution of the Issuer's identity, which may be different than the Client identity.
+1. **Issuer Verification:** The Transparency Service MUST perform resolution of the Issuer's identity, which may be different than the Relying Party identity.
```
```suggestion
1. **Issuer Verification:** The Transparency Service MUST perform resolution of the Issuer's identity.
```

OR13 commented on draft-ietf-scitt-architecture.md:
```diff
@@ -578,7 +578,7 @@ The multiple receipts may be attached to the unprotected header of the Signed St
 ## Transparent Statements {#Receipt}
 
-The client (which is not necessarily the Issuer) that registers a Signed Statement and receives a Receipt can produce a Transparent Statement by adding the Receipt to the Unprotected Header of the Signed Statement.
+The Relying Party (which is not necessarily the Issuer) that registers a Signed Statement and receives a Receipt can produce a Transparent Statement by adding the Receipt to the Unprotected Header of the Signed Statement.
```
```suggestion
Client applications MAY register Signed Statements on behalf of one or more Issuers.
```

OR13 commented on draft-ietf-scitt-architecture.md:
```diff
@@ -833,7 +833,7 @@ All contents exchanged between actors is protected using secure authenticated ch
 The Transparency Service is trusted with the confidentiality of the Signed Statements presented for Registration.
 Some Transparency Services may publish every Signed Statement in their logs, to facilitate their dissemination and auditing.
-Others may just return Receipts to clients that present Signed Statements for Registration, and disclose the Append-only Log only to Auditors trusted with the confidentiality of its contents.
+Others may just return Receipts to Relying Parties that present Singed Statements for Registration, and disclose the Append-only Log only to Auditors trusted with the confidentiality of its contents.
```
```suggestion
Transparency Services MAY return Receipts to client applications synchronously or asynchronously.
```

OR13 commented on draft-ietf-scitt-architecture.md:
```diff
@@ -850,9 +850,9 @@ The SCITT Architecture supports cryptographic agility.
 This enables the gradual transition to stronger algorithms, including e.g. post-quantum signature algorithms.
 
-### Transparency Service Clients
+### Transparency Service Relying Parties
```
```suggestion
### Transparency Service Client Applications
```

OR13 commented on draft-ietf-scitt-architecture.md:
```diff
@@ -850,9 +850,9 @@ The SCITT Architecture supports cryptographic agility.
+### Transparency Service Relying Parties
 
-Trust in clients that submit Signed Statements for Registration is implementation-specific.
+Trust in Relying Parties that submit Signed Statements for Registration is implementation-specific.
```
```suggestion
Authentication of client applications is out of scope for this document.
```

OR13 commented on draft-ietf-scitt-architecture.md:
```diff
@@ -850,9 +850,9 @@ The SCITT Architecture supports cryptographic agility.
 
-Trust in clients that submit Signed Statements for Registration is implementation-specific.
+Trust in Relying Parties that submit Signed Statements for Registration is implementation-specific.
 An attacker may attempt to register any Signed Statement it has obtained, at any Transparency Service that accepts them, possibly multiple times and out of order.
 This may be mitigated by a Transparency Service that enforces restrictive access control and Registration Policies.
```
```suggestion
Transparency Services MUST authenticate both client applications and the Issuer of signed statements in order to ensure that implementation specific authentication and authorization policies are enforced.
The specification of authentication and authorization policies is out of scope for this document.
```

aj-stein-nist commented:
@SteveLasker per discussion, I removed draft state as we want to move this along more quickly.

aj-stein-nist commented:
As I review recommended changes, it appear the current response to this PR is to add definitions of relying party and keep client, which was not the goal of #120 as I interpret it and not my goal in adapting this PR. If that is what is intended, I would rather close this PR. After that, I would like to revisit adapting 120 with feedback from the original author or create a new, focused issue on this and bring up those points in the PR.

At this juncture, the original intent of this PR appears to have become muddled and I have missed the goal of my recommended change. Barring contrary feedback, I will close the PR in the next day or so.

SteveLasker commented on draft-ietf-scitt-architecture.md:
```diff
@@ -188,10 +188,10 @@ Auditor:
 : an entity that checks the correctness and consistency of all Transparent Statements issued by a Transparency Service.
 
-Client:
+Relying Party:
```
```suggestion
Client:
```

SteveLasker commented on draft-ietf-scitt-architecture.md:
```diff
@@ -188,10 +188,10 @@ Auditor:
-: an application making protected Transparency Service resource requests on behalf of the resource owner and with its authorization.
-The term "Client" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other devices).
+: a client application for protected Transparency Service resource requests on behalf of the resource owner and with its authorization.
+The term "Relying Party" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other devices).
```
```suggestion
The term "Client" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other devices).
```

SteveLasker commented:
In an attempt to clarify the replacement of Verifier with Relying Party, while maintaining Client, I've made a PR to this PR: https://github.com/aj-stein-nist/draft-ietf-scitt-architecture/pull/1

To see the combination of AJs #205 with main: https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/compare/main...SteveLasker:steve/120?expand=1
